feat(sdk/java): scaffold Java SDK Gradle multi-module project (#1234)
LFDT-Paladin/paladin: sdk/java/README.md

# Paladin Java SDK

Native Java SDK for [Paladin](https://github.com/LFDT-Paladin/paladin) — enables enterprise

> JVM applications to interact with a Paladin node without going through the Go or TypeScript SDKs.

Reviewer: Suggest slight wording tweak here
```suggestion
JVM applications to interact with a Paladin node using a Java native library.
```

Work in progress...

## Modules

| Module             | Description                                      |
|--------------------|--------------------------------------------------|
| `core`             | Data models and primitives                       |
| `client`           | HTTP + WebSocket transport, RPC client           |
| `domains`          | Helpers for Noto, Pente, and Zeto protocols      |
| `testing`          | Mock client, WireMock stubs, Testcontainers      |
| `integration-test` | End-to-end tests against a real Paladin node     |

## Building

```bash
./gradlew :sdk:java:core:build
./gradlew :sdk:java:client:build
./gradlew :sdk:java:domains:build
./gradlew :sdk:java:testing:build
./gradlew :sdk:java:integration-test:build
```

> Requires JDK 17+.

Reviewer: I think this is worth some discussion. I think your suggestion when we spoke, of Java 21 (in preference to requiring Java 25) was a good one. Perhaps we set Java 21 as the initial pre-req?

Author: ya 21 is good
